packages/executor/src/index.ts

```diff
@@ -1 +1,118 @@
-console.log('Executor module is running...');
\ No newline at end of file
+import market_abi from './abis/market_abi.json'
+import dotenv from 'dotenv'
+import Redis from 'ioredis'
+import chalk from 'chalk'
+import { ethers } from 'ethers'
+
+dotenv.config()
+
+const redis = new Redis({
+  host: process.env.REDIS_HOST,
+  password: process.env.REDIS_PASSWORD,
+})
+
+const log = console.log
+
+interface Position {
+  positionId: string
+  owner: string
+  marketAddress: string
+}
+
+async function liquidatePosition(position: Position) {
+  const { positionId, owner, marketAddress } = position
+
+  try {
+    const privateKeys = process.env.PRIVATE_KEYS?.split(',') ?? []
+
+    const provider = new ethers.providers.JsonRpcProvider(process.env.RPC_URLS?.split(',')[0])
+    const marketContract = new ethers.Contract(marketAddress, market_abi, provider)
+
+    for (const privateKey of privateKeys) {
+      const wallet = new ethers.Wallet(privateKey, provider)
+
+      const [nonce, pendingNonce] = await Promise.all([
+        provider.getTransactionCount(wallet.address, 'latest'),
+        provider.getTransactionCount(wallet.address, 'pending'),
+      ])
```

Comment:
Might be better to keep track internally of which accounts have a pending liquidate tx going on (saves time and RPC calls)

Reply:
Caching the nonces for each wallet could be a good idea but sometimes there are nonce conflicts. For example if the stored nonces are outdated, lost, or if the wallet is used elsewhere.
I had this issues before and that's why I prefer to get the nonce from the provider